README.rst

```diff
@@ -158,6 +158,52 @@ You can also use a :code:`catch_check()` function, if you want to catch an excep
 > Note that some :code:`easycheck` functions are simple wrappers around built-in functions, but their behavior is different, as they have the typical behavior of an :code:`easycheck` function: if a condition is not met, an exception is raised or an issue is raised.
 
 
+Assertions
+-------
+
+In addition to the above checking functions, :code:easycheck provides a set of functions for assertions. They can be used in both code and tests, just like regular assertions using the :code:assert statement. Assertion functions do have a specific functionality that makes them different from the corresponding check functions. You can read more about it here<https://towardsdatascience.com/python-assertions-or-checking-if-a-cat-is-a-dog-ce11c55d143>_. In short, assertions are called only in the development (non-production) mode, that is, when :code:__debug__ is set to :code:True. An assertion should check a condition that should never happen; when the corresponding exception is raised, it means something went wrong in the code, that something that should never happened has just happened.
+
+Some examples:
+
+You are working only on integers, for example pixels when rendering images, or placing objects on a board. You are sure that output will be integer, so you can assert on integers:
+
+.. code-block:: python
+
+    pos.x = 10
+    assert_type(pos.x, int)
+
+Similar situation you will have when you have output from some `len` method:
+
+.. code-block:: python
+
+    out = len("example")
+    assert_type(out, int)
+
+You created the file and worked on it - path assert (note that we are talking about a situation where you have evidence bordering on the certainty that the file exists, you should not use assert if you are not sure if you created the file correctly) e.g. you are working on a README file with example code inside it:
+
+.. code-block:: python
+
+    file_path = "README.rst"
+    assert_paths(file_path)
+
+You are working on subset of some data. So the size of the data should not be larger than the initial one, but also not smaller than 0:
+
+.. code-block:: python
+
+    x = [1,2,3]
+    x1 = x[0:2]
```

Comment:
Instead of this, we should call a function like `subset_data()`, e.g.:

```
def subset(data: pd.DataFrame, filter_condition: Callable) -> pd.DataFrame:
    # create a data frame that is a subset of `data` based on `filter_condition`
    ...
x = pd.DataFrame({'x': [1, 2, 4], 'y': [3, 3, 5]})
x_subset = subset_data(x, lambda value: value < 3)
assert_if_in_limits(len(x1), 0, len(x))
```